src/CMakeLists.txt

```diff
@@ -49,6 +49,9 @@ set(SECP256K1_ENABLE_MODULE_RECOVERY ON CACHE BOOL "" FORCE)
 set(SECP256K1_BUILD_BENCHMARK OFF CACHE BOOL "" FORCE)
 set(SECP256K1_BUILD_TESTS ${BUILD_TESTS} CACHE BOOL "" FORCE)
 set(SECP256K1_BUILD_EXHAUSTIVE_TESTS ${BUILD_TESTS} CACHE BOOL "" FORCE)
+if(NOT BUILD_TESTS)
+  set(SECP256K1_BUILD_CTIME_TESTS ${BUILD_TESTS} CACHE BOOL "" FORCE)
+endif()
```

Comment:
I'm not sure, because in the case being fixed here (`-DBUILD_FOR_FUZZING`), `BUILD_TESTS=OFF`, so we aren't forcing it to `ON`?

It also doesn't explain why our setting (of `OFF`) is being overriden later. The same thing doesn't happen if you pass `SECP256K1_BUILD_CTIME_TESTS=OFF` when invoking cmake.

Reply:
> I'm not sure, because in the case being fixed here (`-DBUILD_FOR_FUZZING`), `BUILD_TESTS=OFF`, so we aren't forcing it to `ON`?

We cannot force it to `ON` because the build will requires the installed Valgrind, which is not always the case.

> It also doesn't explain why our setting (of `OFF`) is being overriden later.

When?

Reply:
When `-DBUILD_FOR_FUZZING=ON` is passed.

Reply:
Could you please suggest your alternative?

Reply:
Something like:
```bash
# Always skip the ctime tests, if we are building no other tests.
# Otherwise, they are built if Valgrind is available. See SECP256K1_VALGRIND.
```
?

Reply:
Thank you! Accepted.